gtk/Session.cc

```diff
@@ -1222,29 +1222,35 @@ void Session::Impl::send_rpc_request(
 
 void Session::port_test()
 {
-    auto const tag = nextTag;
-    ++nextTag;
+    auto const response_func = [this](tr_variant& response)
+    {
+        tr_variant* args = nullptr;
+        auto is_open = false;
+        auto ip_protocol = std::string_view{};
 
-    tr_variant request;
-    tr_variantInitDict(&request, 2);
-    tr_variantDictAddStrView(&request, TR_KEY_method, "port-test");
-    tr_variantDictAddInt(&request, TR_KEY_tag, tag);
-    impl_->send_rpc_request(
-        request,
-        tag,
-        [this](auto& response)
+        if (tr_variantDictFindDict(&response, TR_KEY_arguments, &args))
         {
-            tr_variant* args = nullptr;
-            bool is_open = false;
+            tr_variantDictFindBool(args, TR_KEY_port_is_open, &is_open);
+            tr_variantDictFindStrView(args, TR_KEY_ipProtocol, &ip_protocol);
```

Comment:
Passing back requested protocol type shouldn't be necessary — clients know what they've requested and can use tags to match requests with responses.

Reply:
fd8dd0a GTK
a5aa955 Qt

The RPC response function now will return early if the tag number it got back is not the same with the tag number from the latest request made.

https://github.com/transmission/transmission/blob/a49b806f8a7ae1330feec6295f99e90642bbc2f9/gtk/Session.cc#L1258-L1261

https://github.com/transmission/transmission/blob/a49b806f8a7ae1330feec6295f99e90642bbc2f9/qt/Session.cc#L112-L115